src/main/java/it/pn/frontend/e2e/common/DettaglioNotificaSection.java

```diff
@@ -101,4 +104,73 @@ public void selezioneVediDettaglioButton() {
     public String getTextLinkAttestazioniOpponibili(int i) {
         return attestazioniFile.get(i).getText();
     }
+
+    public void checkTestoBoxPagamento(By xpath, Boolean isTrue) {
+        try {
+            if (isTrue){
+                getWebDriverWait(10).withMessage("campo non trovato").until(ExpectedConditions.visibilityOfElementLocated(xpath));
+            }else {
+                getWebDriverWait(10).withMessage("campo  trovato").until(ExpectedConditions.invisibilityOfElementLocated(xpath));
+            }
+        } catch (TimeoutException e) {
+            logger.error("il box pagamento non é stato caricato correttamente: " + e.getMessage());
+            Assert.fail("il box pagamento non é stato caricato correttamente: " + e.getMessage());
+        }
+    }
```

Comment:
Maybe you can just use the `isDisplayed()` method of the `WebElement`; otherwise, try to minimize the underlying logic.

Reply:
done

Reply:
check the next changes proposed and see if you can apply the same logic